unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,237 @@
+            **{f"{AREA_ID}/response/register_participant": redis_client_auto_register._on_register,
+               f"{AREA_ID}/response/unregister_participant":
+                   redis_client_auto_register._on_unregister,
+               f"{AREA_ID}/*": redis_client_auto_register._on_event_or_response,
+               "aggregator_response": redis_client_auto_register._aggregator_response_callback})
+
+    @staticmethod
+    def test_aggregator_response_callback(redis_client_auto_register):
+        """Check whether the transaction id buffer is popping out and returning the empty list."""
+        data = {"transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._transaction_id_buffer.append(TRANSACTION_ID)
+        redis_client_auto_register._aggregator_response_callback(message)
+        assert redis_client_auto_register._transaction_id_buffer == []
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id(redis_client_auto_register):
+        """Check the buffer message matching the command and id and returns as None."""
+        data = {"transaction_id": TRANSACTION_ID}
+        redis_client_auto_register._blocking_command_responses = (
+            {"register": {"transaction_id": TRANSACTION_ID}})
+        assert redis_client_auto_register._check_buffer_message_matching_command_and_id(
+            data) is None
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id_throws_exception(
+            redis_client_auto_register):
+        """Check exception is raised if the message does not contain transaction_id."""
+        message = {}
+        with pytest.raises(RedisAPIException,
+                           match="The answer message does not contain a valid "
+                                 "'transaction_id' member."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(message)
+
+    @staticmethod
+    def test_check_buffer_message_matching_command_and_id_throws_another_exception(
+            redis_client_auto_register):
+        """Check if the buffer message has no matching command response throws an exception."""
+        data = {"transaction_id": TRANSACTION_ID}
+        with pytest.raises(RedisAPIException,
+                           match="There is no matching command response"
+                                 " in _blocking_command_responses."):
+            redis_client_auto_register._check_buffer_message_matching_command_and_id(data)
+
+    @staticmethod
+    def test_register(redis_client_auto_register):
+        """Check if the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.register(is_blocking=False)
+        transaction_id = redis_client_auto_register._blocking_command_responses[
+            "register"]["transaction_id"]
```

Comment:
Since `_blocking_command_responses` is a private property, we should not use it in our tests.
You're using it because you want to know the value of `transaction_id` (since you need it for the assertion later on). 
What you can do is treating `uuid.uuid4()` as what it really is: an outgoing query to an external function. You can just patch it in this test and use any string (e.g. `"some-transaction-uuid"`) as its return value.

Reply:
Sure, I will adapt these changes.